spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -129,6 +130,17 @@ public void setPhysicalCloseTimeout(int physicalCloseTimeout) {
 	public void setTransactionIdPrefix(String transactionIdPrefix) {
 		Assert.notNull(transactionIdPrefix, "'transactionIdPrefix' cannot be null");
 		this.transactionIdPrefix = transactionIdPrefix;
+		enableIdempotentBehaviour();
+	}
+
+	/**
+	 * When set to 'true', the producer will ensure that exactly one copy of each message is written in the stream.
+	 */
+	private void enableIdempotentBehaviour() {
+		Object previousValue = this.configs.putIfAbsent(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true);
+		if (logger.isDebugEnabled() && previousValue != null && previousValue.equals(false)) {
+			logger.debug(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG +  "is set to " + previousValue + ", may result in duplicate messages");
```

Comment:
The `ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG` is resolved to the `enable.idempotence` string and then you have an `is` word. But there is no whitespace between them.

I would wrap `ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG` to the single quotes and add `The ` in the beginning of sentence.